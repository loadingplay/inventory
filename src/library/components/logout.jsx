import React from 'react'


export class LogoutButton extends React.PureComponent {

  handleClick = () => {
    localStorage.setItem("access_token", "");
    document.location.href = `https://accounts.loadingplay.com/auth/logout?next=${process.env.PROJECT_URL}`;
  }

  render = () => {
    return (
      <button
        onClick={this.handleClick}
        role="button"
        className="btn btn-link ml-1" >
        Salir
      </button>
    );
  }
}
