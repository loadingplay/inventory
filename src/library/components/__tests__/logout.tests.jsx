import React from "react";
import { shallow } from 'enzyme';

import { LogoutButton } from "../logout";

describe("Logout", () => {

  const OLD_ENV = process.env;

  beforeEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
    process.env = {...OLD_ENV};
  });

  afterEach(() => {
    process.env = {...OLD_ENV};
  })

  test("it should render logout button", () => {

    process.env.PROJECT_URL = "http://test-url";

    const wrapper = shallow(
      <LogoutButton />,
      {
        disableLifecycleMethods: true
      }
    );
    expect(wrapper).toMatchSnapshot("default");
  });
});
