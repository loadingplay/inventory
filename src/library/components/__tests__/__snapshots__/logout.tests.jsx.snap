// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout it should render logout button: default 1`] = `
<button
  className="btn btn-link ml-1"
  onClick={[Function]}
  role="button"
>
  Salir
</button>
`;
